hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/spark/sql/hudi/TestCreateTable.scala

```diff
@@ -763,4 +763,22 @@ class TestCreateTable extends HoodieSparkSqlTestBase {
       assertResult(true)(shown.contains("COMMENT 'This is a simple hudi table'"))
     }
   }
+
+  test("Test CTAS using an illegal definition -- a COW table with compaction enabled.") {
+    val tableName = generateTableName
+    checkExceptionContain(
+      s"""
+         | create table $tableName using hudi
+         | tblproperties(
+         |    primaryKey = 'id',
+         |    type = 'cow',
+         |    hoodie.compact.inline='true'
```

Comment:
sorry I do not see the check logic for the invalid definition in this PR, am I missing it?

Reply:
You mean this line ?https://github.com/apache/hudi/pull/5588/files/c06740700f51f557b77ffc19068e36a4cb19864a#diff-52cf34bac0cb6cc5b9c0f46c5fd190ac202a1db824182b59137d83ae0db38c89R779
"Compaction is not supported on a CopyOnWrite table"

Reply:
no, what i mean is the check logic in hudi

Reply:
Current HoodieSparkCopyOnWriteTable #scheduleCompaction and #compact throws `new HoodieNotSupportedException("Compaction is not supported on a CopyOnWrite table");`

Reply:
@jinxing64 but how did the create table with `hoodie.compact.inline=true` call `HoodieSparkCopyOnWriteTable #scheduleCompaction` or `#compact`? ps: you can add my wechat(xleesf) to communicate offline

Reply:
@jinxing64 but how did the create table with `hoodie.compact.inline=true` call `HoodieSparkCopyOnWriteTable #scheduleCompaction` or `#compact`? ps: you can add my wechat(xleesf) to communicate offline

Reply:
I applied on "wechat" :)
Stacktrace as below:
```
org.apache.hudi.exception.HoodieNotSupportedException: Compaction is not supported on a CopyOnWrite table
	at org.apache.hudi.table.HoodieSparkCopyOnWriteTable.scheduleCompaction(HoodieSparkCopyOnWriteTable.java:169)
	at org.apache.hudi.client.BaseHoodieWriteClient.scheduleTableServiceInternal(BaseHoodieWriteClient.java:1348)
	at org.apache.hudi.client.BaseHoodieWriteClient.scheduleTableService(BaseHoodieWriteClient.java:1325)
	at org.apache.hudi.client.BaseHoodieWriteClient.scheduleCompactionAtInstant(BaseHoodieWriteClient.java:1003)
	at org.apache.hudi.client.BaseHoodieWriteClient.scheduleCompaction(BaseHoodieWriteClient.java:994)
	at org.apache.hudi.client.BaseHoodieWriteClient.inlineScheduleCompaction(BaseHoodieWriteClient.java:1252)
	at org.apache.hudi.client.BaseHoodieWriteClient.inlineCompaction(BaseHoodieWriteClient.java:1238)
	at org.apache.hudi.client.BaseHoodieWriteClient.runTableServicesInline(BaseHoodieWriteClient.java:552)
	at org.apache.hudi.client.BaseHoodieWriteClient.commitStats(BaseHoodieWriteClient.java:245)
	at org.apache.hudi.client.SparkRDDWriteClient.commit(SparkRDDWriteClient.java:122)
	at org.apache.hudi.HoodieSparkSqlWriter$.commitAndPerformPostOperations(HoodieSparkSqlWriter.scala:651)
	at org.apache.hudi.HoodieSparkSqlWriter$.write(HoodieSparkSqlWriter.scala:315)
	at org.apache.hudi.DefaultSource.createRelation(DefaultSource.scala:163)
	at org.apache.spark.sql.execution.datasources.SaveIntoDataSourceCommand.run(SaveIntoDataSourceCommand.scala:45)
```

Reply:
In my understanding about the compaction triggering mechanism -- when inlineCompaction is enabled, `HoodieSparkTable#scheduleCompaction` is invoked from `HoodieWriteClient`. Then `ScheduleCompactionActionExecutor` decides the triggering timing;

The exception above (`Compaction is not supported`) is thrown from `HoodieSparkCopyOnWriteTable`